sources/imdsclient/src/lib.rs

```diff
@@ -21,20 +21,27 @@ use log::{debug, info, trace, warn};
 use reqwest::Client;
```

Comment:
I think it'd be worth having a doc comment or something explaining the various timeout values involved here.  I see 60 seconds for the retry strategy, 300 seconds for the retry token timeout (which I think means we'd run the retry process ~5 times?), and 1 second between non-token IMDS request attempts.  It would be helpful to document where those numbers come from and how they interact.  (They all seem high to me.)

Reply:
There are a couple comments in the code now, but it seems to me that the retries and delay are an important part of usage of the library, and since they're not all configurable and are tied to a somewhat arbitrary timeout in another program, I think a comment of higher prominence could be helpful - right in the module documentation perhaps?

Reply:
I was a tad worried if the comments were too far away from the code the timings could potentially be updated without making the appropriate changes to the documentation. I could add blurb in the module documentation that describes the behavior, but doesn't called out the exact timings if that helps.

Reply:
> I could add blurb in the module documentation that describes the behavior

That seems reasonable to me. I think it is unusual to retry something for several minutes, so a bit of a prominent documentation of this makes sense.